src/app/modules/auth/sgid/auth-sgid.service.ts

```diff
@@ -0,0 +1,122 @@
+import { SgidClient } from '@opengovsg/sgid-client'
+import fs from 'fs'
+import { err, ok, Result, ResultAsync } from 'neverthrow'
+
+import { ISgidVarsSchema } from 'src/types'
+
+import { sgid } from '../../../config/features/sgid.config'
+import { createLoggerWithLabel } from '../../../config/logger'
+import {
+  SgidCreateRedirectUrlError,
+  SgidFetchAccessTokenError,
+  SgidFetchUserInfoError,
+} from '../../sgid/sgid.errors'
+
+const logger = createLoggerWithLabel(module)
+
+export const SGID_LOGIN_OAUTH_STATE = 'login'
+const SGID_OGP_WORK_EMAIL_SCOPE = 'ogpofficerinfo.work_email'
+
+export class AuthSgidServiceClass {
+  private client: SgidClient
+  private privateKey: string
+
+  constructor({
+    privateKeyPath,
+    hostname,
+    adminLoginRedirectUri: redirectUri,
+    clientId,
+    clientSecret,
+  }: ISgidVarsSchema) {
+    this.privateKey = fs.readFileSync(privateKeyPath, { encoding: 'utf8' })
+    this.client = new SgidClient({
+      // If hostname is empty, use the default provided by sgid-client.
+      hostname: hostname || undefined,
+      clientId,
+      clientSecret,
+      redirectUri,
+      privateKey: this.privateKey,
+    })
+  }
+
+  /**
+   * Create a URL to sgID which is used to redirect the user for authentication
+   */
+  createRedirectUrl(): Result<string, SgidCreateRedirectUrlError> {
+    const logMeta = {
+      action: 'createRedirectUrl',
+    }
+
+    const result = this.client.authorizationUrl(
+      SGID_LOGIN_OAUTH_STATE,
+      ['openid', SGID_OGP_WORK_EMAIL_SCOPE].join(' '),
+      null,
+    )
+    if (typeof result.url === 'string') {
```

Comment:
nit: redundant check

Reply:
ahh true. Switched to try-catch instead. Thanks!